cql3/statements/index_target.cc

```diff
@@ -140,7 +140,6 @@ sstring to_sstring(index_target::target_type type)
     case index_target::target_type::collection_values: return "values";
     case index_target::target_type::full: return "full";
     }
-    return "";
```

Comment:
This is needed for gcc (which we want to move back to one day). It believes enums can contain values other than those declared, and so it thinks this isn't dead (and complains if it's missing). Can be replaced by on_internal_error().

Reply:
There's no logger there, so perhaps I'll do what other functions do in that function, i.e. throw a runtime error.

Reply:
Don't know why github threaded the message in the way it did.

Reply:
Fair enough. @enedil I suggest the following - let's try to speedrun the implementation as far as you can, leaving the trivial/boilerplate bits for now. E.g. ensuring that the code is properly guarded by cluster features, that rolling upgrades work, what's the exact schema of the new table, can be left for later, for now just try to make sure that the new code works on current master. Then, when June 18th comes, we'll hopefully be in a state where the code is more or less working, but needs a few relatively easy change to be applied, and that can be picked up by someone else (e.g. me). If you don't manage to finish a working implementation by June 18th, spend the last day on thoroughly documenting what's left to be done and what are the expected pitfalls/corner cases that you expect to be handled but didn't manage to.